### ISSUE TITLE
Default `parserOptions.ecmaFeatures.jsx: true` breaks type assertion parsing in setup scripts

### ISSUE THREAD

**Issue by dmolesUC** [vuejs/eslint-config-typescript#55]
## Steps to reproduce:

1. Clone [dmolesUC/ts-test](https://github.com/dmolesUC/ts-test) for a complete reproducible example, or:

   1. create a Vue3/TypeScript project by running `yarn create vite` and selecting `Vue` and `TypeScript`
   2. add and configure ESLint, including this plugin
   3. modify `components/HelloWorld.vue` to contain the code below:

      ```vue
      <script setup lang="ts">
      import { Ref, ref } from 'vue'

      defineProps<{ msg: string }>()

      const count: Ref<number | undefined> = ref(0)

      function incrementCount() {
        const c = <number>count.value // <- type assertion is here
        count.value = c + 1
      }
      </script>
      ```

2. Run `yarn eslint --ext .js,.ts,.vue src`

## Expected

- Code lints without errors; or, at least, ESLint is able to parse the Vue component source file.

## Actual

- Closing brace of the `incrementCount()` function produces the following parsing error:

   ```none
   /Users/david/Work/Scratch/ts-test/src/components/HelloWorld.vue
     12:0  error  Parsing error: Unexpected token. Did you mean `{'}'}` or `&rbrace;`?

   ✖ 1 problem (1 error, 0 warnings)
   ```

## Workaround

- in `.eslintrc.cjs`, add `ecmaFeatures: { jsx: false }` to `parserOptions`

## Note

I originally filed this as https://github.com/vuejs/vue-eslint-parser/issues/177, but @ota-meshi identified the issue as being the `jsx` config in [`eslint-config-typescript/index.js`](https://github.com/vuejs/eslint-config-typescript/blob/v11.0.2/index.js#L20).

It's not clear to me why JSX support needs to be enabled by default for Vue projects, which I think tend not to use it that often. But at least this issue documents the workaround.
